File: zombie-survivor/csharp/src/ZombieSurvivor/Exceptions.cs
namespace ZombieSurvivor;

public class EquipmentCapacityExceededException : Exception
{
    public EquipmentCapacityExceededException(string message) : base(message) { }
}

public class DuplicateSurvivorNameException : Exception
{
    public DuplicateSurvivorNameException(string message) : base(message) { }
}

File: zombie-survivor/typescript/src/Exceptions.ts
export class EquipmentCapacityExceededException extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'EquipmentCapacityExceededException';
  }
}

export class DuplicateSurvivorNameException extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'DuplicateSurvivorNameException';
  }
}
